pkg/web/apps/apps_psql_repository.go

```diff
@@ -106,3 +111,19 @@ func (a *appsPostgreSQLRepository) GetAppVersionsByAppID(id string) (*[]models.V
 
 	return &versions, nil
 }
+
+// GetAppByID retrieves an app by id from the database
+func (a *appsPostgreSQLRepository) GetAppByID(ID string) (*models.App, error) {
+	var app models.App
+	//var v models.Version
+
+	sqlStatment := `SELECT id,app_id,app_name FROM app WHERE id=$1;`
+	row := a.db.QueryRow(sqlStatment, ID)
+	err := row.Scan(&app.ID, &app.AppID, &app.AppName)
+	if err != nil {
+		return nil, err
```

Comment:
If you return the error `models.ErrNotFound` here then you won't need to do a check in the service layer.

Reply:
agreed with this. The code will be:
```
	if err != nil {
		log.Error(err)
		if err == sql.ErrNoRows {
			return nil, models.ErrNotFound
		}
		return nil, models.ErrInternalServerError
	}
```

Reply:
:+1: